src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/EntitySchemaProcessor.cs

```diff
@@ -151,12 +173,48 @@ public void ProcessChanges()
                     EntityId = lookup.EntityId,
                     SchemaHash = entry.Key.SchemaHash.ConvertFromHex(),
                     Schema = entry.Value.Schema.SborData.Hex.ConvertFromHex(),
+                    IsDeleted = false,
                 };
 
                 _definitionsToAdd.Add(entryDefinition);
 
                 aggregate.EntryIds.Insert(0, entryDefinition.Id);
             }
+
+            foreach (var deletedSchemaHash in change.DeletedSchemaHashes)
+            {
+                var entryLookup = new SchemaDefinitionEntryDbLookup(lookup.EntityId, deletedSchemaHash.ConvertFromHex());
+
+                if (_schemasToBeDeleted.TryGetValue(entryLookup, out var previousEntry))
+                {
+                    var currentPosition = aggregate.EntryIds.IndexOf(previousEntry.Id);
+
+                    if (currentPosition != -1)
+                    {
+                        aggregate.EntryIds.RemoveAt(currentPosition);
+                    }
+                    else
+                    {
+                        throw new UnreachableException($"Unexpected situation where SchemaEntryDefinition with EntityId:{entryLookup.EntityId}, SchemaHash:{deletedSchemaHash} got deleted but wasn't found in aggregate table.");
```

Comment:
Are we 100% sure schema cannot be deleted on its own somehow? Be advised schemas are not limited to package blueprints.